Make generate-dirty-files and compare-static-analysis-results work with iOS

rniwa opened the pull request:
#### 31050dea93207bce9c9f7139936233b0b89880a7
<pre>
Make generate-dirty-files and compare-static-analysis-results work with iOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=300377">https://bugs.webkit.org/show_bug.cgi?id=300377</a>

Reviewed by Chris Dumez.

Extend generate-dirty-files and compare-static-analysis-results so that we can add
static analyzer expectations for iOS. We do by allowing micro-syntax similar to the one
used in TestExpectations in the expectation files; e.g. `[ iOS ] FileWithIssue.cpp`

Also strip WebKitBuild/Release/ or WebKitBuild/Release-iphonesimulator/ from the file
path when generating a &quot;dirty&quot; list of files so that the results can be shared across
platforms.

* Source/WTF/SaferCPPExpectations/ForwardDeclCheckerExpectations:
* Source/WTF/SaferCPPExpectations/NoUnretainedMemberCheckerExpectations:
* Source/WTF/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Tools/Scripts/compare-static-analysis-results:
(parser):
(find_diff):
* Tools/Scripts/generate-dirty-files:
(parse_results_file):

Canonical link: <a href="https://commits.webkit.org/301254@main">https://commits.webkit.org/301254@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fa47d7aa0c9034e293515368327f7c416d9106f1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125320 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44985 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35726 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132169 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77189 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/56cdd166-ec2c-4cc6-be7b-1736523da221) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/127197 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45673 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53550 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95414 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63343 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/41b22c11-df79-449d-93fa-ac02814a916c) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128274 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36473 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112069 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75953 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/aa81fb80-0341-4580-bde1-917577cd98de) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35372 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30238 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75646 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/117410 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106241 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30464 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134855 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/123838 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52123 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39901 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103887 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52561 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108291 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103648 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49007 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27302 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/49233 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19633 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52018 "Built successfully") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57798 "Found 3048 new failures in DerivedSources/WebCore/JSDeprecatedCSSOMValueList.cpp, DerivedSources/WebCore/JSDOMPlugin.cpp, DerivedSources/WebKit/JSWebExtensionAPIDeclarativeNetRequest.mm, DerivedSources/WebCore/JSNVShaderNoperspectiveInterpolation.cpp, DerivedSources/WebCore/JSDatabaseCallback.cpp, DerivedSources/WebCore/JSShadowRealmGlobalScope.cpp, DerivedSources/WebCore/JSTrustedTypePolicyFactory.cpp, DerivedSources/WebCore/JSWebGLFramebuffer.cpp, DerivedSources/WebCore/JSWebGLCompressedTexturePVRTC.cpp, DerivedSources/WebCore/JSErrorCallback.cpp ...") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/156857 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51375 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39279 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54731 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53069 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/79baea9089df6a7013b4aecd1900d3da222648bb. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51995

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7fefb648a94987b10fea7f0279b35e860bfa35c5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51995

rniwa committed:
Make generate-dirty-files and compare-static-analysis-results work with iOS
https://bugs.webkit.org/show_bug.cgi?id=300377

Reviewed by Chris Dumez.

Extend generate-dirty-files and compare-static-analysis-results so that we can add
static analyzer expectations for iOS. We do by allowing micro-syntax similar to the one
used in TestExpectations in the expectation files; e.g. `[ iOS ] FileWithIssue.cpp`

Also strip WebKitBuild/Release/ or WebKitBuild/Release-iphonesimulator/ from the file
path when generating a "dirty" list of files so that the results can be shared across
platforms.

* Source/WTF/SaferCPPExpectations/ForwardDeclCheckerExpectations:
* Source/WTF/SaferCPPExpectations/NoUnretainedMemberCheckerExpectations:
* Source/WTF/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Tools/Scripts/compare-static-analysis-results:
(parser):
(find_diff):
* Tools/Scripts/generate-dirty-files:
(parse_results_file):

Canonical link: https://commits.webkit.org/301254@main
